libs/sdk-ui-kit/src/Dialog/ConversionTouchPointDialogBase.tsx

```diff
@@ -0,0 +1,189 @@
+    [key: string]: string | number | boolean;
+}
+
+/**
+ * @public
+ */
+export interface IConversionTouchPointDialogBaseProps {
+    /**
+     * Hubspot Portal Id
+     */
+    hubspotPortalId: string;
+
+    /**
+     * Hubspot Form Id
+     */
+    hubspotFormId: string;
+
+    /**
+     * Dialog Title
+     */
+    dialogTitle?: string;
+
+    /**
+     * The value for Cancel button
+     */
+    cancelButtonText?: string;
+
+    /**
+     * Show/Hide Cancel button
+     */
+    showCancelButton?: boolean;
+
+    /**
+     * Populate values for the Hubspot form
+     */
+    values?: IHubspotFormValue;
+
+    /**
+     * Mark the checkbox as checked in the Hubspot form base on this value.
+     */
+    selectedValue?: string;
+
+    /**
+     * The id for html element render the Hubspot form
+     */
+    targetId?: string;
+
+    /**
+     * Close dialog action
+     */
+    onClose: () => void;
+
+    /**
+     * Form submitted callback function
+     */
+    onFormSubmitted?: () => void;
+}
+
+/**
+ * @internal
+ */
+export const ConversionTouchPointDialogBase: React.FC<IConversionTouchPointDialogBaseProps> = (props) => {
+    const {
+        targetId,
+        hubspotPortalId,
+        hubspotFormId,
+        selectedValue,
+        cancelButtonText,
+        dialogTitle,
+        showCancelButton,
+        values = {},
+        onClose,
+        onFormSubmitted,
+    } = props;
+
+    const intl = useIntl();
+    const [isFormReady, setIsFormReady] = useState(false);
+    const hubspotFormTargetId = targetId || "conversion-touch-point-hubspot";
+
+    useEffect(() => {
+        // the Hubspot form requires jQuery object, we need to create a fake jQuery object to bypass this.
+        window.jQuery =
+            window.jQuery ||
+            function (nodeOrSelector: string | object) {
+                if (typeof nodeOrSelector == "string") {
+                    return document.querySelector(nodeOrSelector);
+                }
+                return nodeOrSelector;
+            };
+        if (window.hbspt?.isSuccessMessageShow) {
+            window.hbspt.isSuccessMessageShow = false;
+        }
+    }, []);
+
+    const onHubspotFormSubmitted = () => {
+        if (!window.hbspt.isSuccessMessageShow) {
+            onFormSubmitted?.();
+            onClose();
+            window.hbspt.isSuccessMessageShow = true;
+        }
+    };
+
+    const { formCreated } = useHubspotForm({
+        portalId: hubspotPortalId,
+        formId: hubspotFormId,
+        target: `#${hubspotFormTargetId}`,
+        locale: intl.locale.split("-").shift() as HubSpotFormLocale,
+        onFormSubmitted: onHubspotFormSubmitted,
+        onFormReady: ($form: IHubspotFormField[]) => {
+            setIsFormReady(true);
+            // populating the values for hidden fields
+            for (let i = 0; i <= $form.length; i += 1) {
+                const inputField = $form[i];
+                if (!inputField) {
+                    return;
```

Comment:
Just a confirmation: if `inputField` is null/undefined for an index `i`, can we be sure that `$form[i]` is also null/undefined for all remaining indexes (`i + 1`, `i + 2`, ..., `$form.length - 1`)?

Reply:
The index should be from 0 to $form.length - 1. I have updated the condition to make i always smaller than $form.length.

Reply:
thanks, but the main point I want to confirm with you is that, should we replaced `return;` by `continue;`?
※ We can keep `return` if we can sure that if `$form[i]` is null/undefined, then `$form[i + 1]`, `$form[i + 2]`, ... is also null/undefined.

Reply:
yes, I have replace `return` by `continue`. Thanks!